packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -1426,6 +1429,14 @@ export class UserPool extends UserPoolBase {
     if (props.allowedFirstAuthFactors.passkey) {
       allowedFirstAuthFactors.push('WEB_AUTHN');
     }
+    if (allowedFirstAuthFactors.length === 1) {
+      return undefined;
+    }
```

Comment:
This could introduce subtle bugs in case the number of required auth methods changes for some reason.
Let's check `!Object.keys(allowedFirstAuthFactors).length` at L1417 instead.

Reply:
Thank you for your suggestion.

But I'll revert this because:
1. Create a stack with a user pool including `SignInPolicy: ['PASSWORD','WEB_AUTHN']`
2. Remove the `SignInPolicy` then update the stack
3. Management console shows the user pool still enables choice-based sign-in with Password and Passkey
4. Set `SignInPolicy: ['PASSWORD']` then update the stack again
5. Management console shows the user pool disables choice-based sign-in.

Therefore, we should provide the way to disable choice-based sign-in explicitly by specifying `allowedFirstAuthFactors: {}`.

Reply:
Similar issue: #30796

Reply:
I see, thanks for checking. 
However, allowing for `allowedFirstAuthFactors: {}` makes for an unclear implementation.
`allowedFirstAuthFactors: { password: true }` with `password` required to `true` could be an option to avoid that, even if that would mean adding a new validation rule.

Reply:
Agreed. I'll add.